### PR TITLE
feat(functions): New v2 sample for GCE labeling on new instance creation

### PR DESCRIPTION
**Pull request by muncus**
- feat(functions): gcfv2 sample for GCE audit logs.
- add a check for malformed subject.

### CODE REVIEW

File: functions/functionsv2/label_gce_instance/go.mod
Comment (grayside) [inline functions/functionsv2/label_gce_instance/go.mod:17]: Not familiar with this syntax using multiple require blocks. Perhaps a `go mod tidy` is in order?

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (grayside) [inline functions/functionsv2/label_gce_instance/label_gce_instance.go:29]: Very rare to see gax-go as a sample dependency. Curious to see where it's used as I continue the review.

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (grayside) [inline functions/functionsv2/label_gce_instance/label_gce_instance.go:78]: suggestion: Given how common it is to use `e` as a secondary name for `err`, maybe we should change the variable representing event to `ce` or `ev`?

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (grayside) [inline functions/functionsv2/label_gce_instance/label_gce_instance.go:96]: e.g., "label sanitization"?

File: functions/functionsv2/label_gce_instance/label_gce_instance_test.go
Comment (grayside) [inline functions/functionsv2/label_gce_instance/label_gce_instance_test.go:30]: Thanks for putting together a thorough unit test!

Generally, we prefer integration tests (or extra credit, [e2e tests](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/functions/helloworld/hello_http_system_test.go)). These lead to the test doubling as a check on overall sample reliability.

Identifying how to consistently meet that standard in CloudEvent functions samples is forthcoming, so this is not a blocking concern but I wanted to flag as a specific case since this testing approach is affecting the sample UX.

**Review by grayside — APPROVED** [review]
Looking great, left a few points of feedback but nothing blocking. We may want to request GCE DPE product review before merging.

Process Note: Title the PR using conventionalcommits.org, amongst other things this helps the automation identify which product to use. New samples generally `feat(product-short-name): Short sample explanation`

File: functions/functionsv2/label_gce_instance/go.mod
Comment (muncus) [reply functions/functionsv2/label_gce_instance/go.mod:17]: This was not an intentional choice, and tidy does not merge the two blocks.
I've merged them manually and everything seems fine.

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (muncus) [reply functions/functionsv2/label_gce_instance/label_gce_instance.go:78]: good call. fixed.

File: functions/functionsv2/label_gce_instance/label_gce_instance_test.go
Comment (muncus) [reply functions/functionsv2/label_gce_instance/label_gce_instance_test.go:30]: I agree that this approach to testing is not great for the sample readability and UX (though it is one of the [recommended approaches](https://github.com/googleapis/google-cloud-go/blob/main/testing.md#testing-using-mocks). )

I see at least the following options, and am open to others:
We could leave the tests as-is, or ( to prioritize the sample UX over the testing) switch to just testing the function's return value.
How do you suggest we proceed?

Comment (codyoss) [reply functions/functionsv2/label_gce_instance/label_gce_instance_test.go:30]: I think it may be cleaner to actually use a fake here instead of a mock for the sake of the sample, see the first section in the doc you linked.  Then we don't need to define an interface in the sample at all and can just work with the real client.

Comment (muncus) [reply functions/functionsv2/label_gce_instance/label_gce_instance_test.go]: I discussed this with Cody, and it turns out the GCE services require an [http api approach](https://github.com/googleapis/google-api-go-client/blob/main/testing.md#testing-http-services-using-fakes), which is different from the grpc guidance I initially linked.
I've updated the sample tests to use a fake server.

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (codyoss) [inline functions/functionsv2/label_gce_instance/label_gce_instance.go:64]: nit: perhaps just shadow err here.

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (codyoss) [inline functions/functionsv2/label_gce_instance/label_gce_instance.go:65]: nit: I am not sure we need these extra logs where errors are returned. Seem duplicative.

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (codyoss) [inline functions/functionsv2/label_gce_instance/label_gce_instance.go:70]: nit: on line below consider err for naming(github wont let me comment on the line below right now...)

File: functions/functionsv2/label_gce_instance/label_gce_instance_test.go
Comment (codyoss) [inline functions/functionsv2/label_gce_instance/label_gce_instance_test.go:44]: Since we are working with generated proto stubs we should use the protojson library: https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson

**Review by codyoss — CHANGES_REQUESTED** [review]
Just a couple small things, or else it LGTM. Thanks for refactoring the testing!

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (muncus) [reply functions/functionsv2/label_gce_instance/label_gce_instance.go:64]: Done here, and in a similar case below on line 99

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (muncus) [reply functions/functionsv2/label_gce_instance/label_gce_instance.go]: That was my initial though as well, but in my experience, just returning the error gets you a log message that the response was a 500, but does not include the string representation of the returned error.
I think the added logging is worthwhile, since it provides necessary debugging info for users of the sample.

File: functions/functionsv2/label_gce_instance/label_gce_instance_test.go
Comment (muncus) [reply functions/functionsv2/label_gce_instance/label_gce_instance_test.go]: Thanks, i did not know about the protojson library. done.

File: functions/functionsv2/label_gce_instance/label_gce_instance.go
Comment (codyoss) [reply functions/functionsv2/label_gce_instance/label_gce_instance.go]: Ack, we can do it here if you feel strongly, but this is not something we have been doing in general with our samples.

**Review by codyoss — APPROVED** [review]

**Pull request merged**
